uwrt_arm_cartesian_controller/src/uwrt_arm_cartesian_controller.cpp

```diff
@@ -0,0 +1,224 @@
+
+    CallbackReturn UWRTCartesianController::on_deactivate(
+        const rclcpp_lifecycle::State & /*previous_state*/)
+    {
+        for (auto &command_interface : command_interfaces_)
+        {
+            command_interface.set_value(0.0);
+        }
+
+        m_last_twist = get_null_twist();
+
+        return CallbackReturn::SUCCESS;
+    }
+
+    controller_interface::return_type UWRTCartesianController::update(const rclcpp::Time &time, const rclcpp::Duration & /*period*/)
+    {
+        // expired command
+        if (time - m_last_twist.header.stamp > rclcpp::Duration(5, 0)) // younes todo make this a param
+        {
+            KDL::SetToZero(m_cart_vel);
+            for (auto &command_interface : command_interfaces_)
+            {
+                command_interface.set_value(0.0);
+            }
+        }
+        else
+        {
+            // younes todo IK magic here
+        }
+
+        return controller_interface::return_type::OK;
+    }
+
+    bool UWRTCartesianController::parse_params()
+    {
+        m_joint_names = node_->get_parameter("joints").as_string_array();
+
+        if (m_joint_names.empty())
+        {
+            RCLCPP_ERROR(get_node()->get_logger(), "'joints' parameter was empty");
+            return false;
+        }
+
+        // Loop over string params
+        const std::array<std::pair<std::reference_wrapper<std::string>, std::string>, 5> string_params = {{m_interface_name, "interface_name"},
+                                                                                                          {m_cart_command_topic, "end_effector_velocity_topic"},
+                                                                                                          {m_robot_description, "robot_description"},
+                                                                                                          {m_root_name, "root_name"},
+                                                                                                          {m_tip_name, "tip_name"}};
+        for (auto &string_param : string_params)
+        {
+            if (string_param.first.empty())
+            {
+                string_param.first = node_->get_parameter(string_param.second).as_string();
+                if (string_param.first.empty())
+                {
+                    RCLCPP_ERROR(get_node()->get_logger(), string_param.second + " parameter was empty");
+                    return false;
+                }
+            }
+        }
+
+        return true;
+    }
+
+    TwistStamped UWRTCartesianController::get_null_twist() const
+    {
+        TwistStamped ret;
+        ret.twist.linear.x = 0;
+        ret.twist.linear.y = 0;
+        ret.twist.linear.z = 0;
+        ret.twist.angular.x = 0;
+        ret.twist.angular.y = 0;
+        ret.twist.angular.z = 0;
+        ret.header.stamp = rclcpp::Time::now();
+        // younes todo whats the name of the arm's base frame ret.header.frame_id = "foo";
```

Comment:
see #173

Reply:
it is rn. it should be changed. @meshvaD

Reply:
Why do you actually need to hard code link names? They could be loaded ad parameters (ex. End effector link and arm base link)